get_prism_data.R

```diff
@@ -4,39 +4,232 @@ library(prism)
+  prism_bbs_data=get_prism_data()
+  
+  #Spread out the climate variables ppt, tmean, etc into columns
+  prism_bbs_data = prism_bbs_data %>%
+    spread(clim_var, value)
+  
+  #Process the quarter ones first.
+  quarterInfo=data.frame(month=1:12, quarter=c(1,1,1,2,2,2,3,3,3,4,4,4))
+  bioclimQuarterData= prism_bbs_data %>%
+    left_join(quarterInfo, by='month') %>%
+    group_by(site_id, year, quarter) %>%
+    summarize(precip=sum(ppt), temp=mean(tmean)) %>%
+    ungroup() %>%
+    group_by(site_id,year) %>%
+    summarize(bio8=maxMinCombo(temp, precip, max=TRUE),
+              bio9=maxMinCombo(temp, precip, max=FALSE),
+              bio10=max(temp),
+              bio11=min(temp),
+              bio16=max(precip),
+              bio17=min(precip),
+              bio18=maxMinCombo(precip, temp, max=TRUE),
+              bio19=maxMinCombo(precip, temp, max=FALSE)) %>%
+    ungroup()
+  
+  #Next the yearly ones, joining the quartely ones  back in at the end. 
+  bioclimData=prism_bbs_data %>%
+    group_by(site_id, year) %>%
+    mutate(monthlyTempDiff=tmax-tmin) %>%
+    summarize(bio1=mean(tmean),
+              bio2=mean(monthlyTempDiff),
+              bio4=sd(tmean)*100,
+              bio5=maxMinCombo(tmax,tmean,max=TRUE),
+              bio6=maxMinCombo(tmin,tmean,max=FALSE),
+              bio12=sum(ppt),
+              bio13=max(ppt),
+              bio14=min(ppt),
+              bio15=cv(ppt)) %>%
+    ungroup() %>%
+    mutate(bio7=bio5-bio6,
+           bio3=(bio2/bio7)*100) %>%
+    full_join(bioclimQuarterData, by=c('site_id','year'))
+  
+  return(bioclimData)
 }
 
-bbs_data <- read.csv("data/bbs_data.csv")
-locations <- unique(select(bbs_data, site_id, long, lat))
-coordinates(locations) <- c("long", "lat")
-#prism_files <- grep("_[0-9]{4}[0]",ls_prism_data()[,1],value=T)
-prism_stacked <- prism_stack(ls_prism_data())
-extracted <- raster::extract(prism_stacked, locations)
-prism_bbs_data <- data.frame(site_id = locations$site_id, coordinates(locations), extracted)
-prism_bbs_data <- prism_bbs_data %>%
-                  gather(date, value, 4:ncol(prism_bbs_data)) %>%
-                  tidyr::extract(date, c("clim_var", "year", "month"),
-                                 "PRISM_([:alpha:]*)_stable_[:alnum:]*_([:digit:]{4})([:digit:]{2})_")
-prism_bbs_data$year <- as.numeric(prism_bbs_data$year)
-prism_bbs_data$month <- as.numeric(prism_bbs_data$month)
-database <- src_sqlite("./data/bbsforecasting.sqlite", create = TRUE)
-mydata <- copy_to(database, prism_bbs_data, temporary = FALSE,
-                  indexes = list(c("site_id", "year", "month")))
+
+####################################################################
+#Load the bioclim variables from sqlite. if they aren't available,
+#load the prism data, process bioclim, load it in sqlite, then return 
+#bioclim as requested.
+
+#Possibly put a check here to make sure all years are accounted for?
```

Comment:
I generally recommend starting comments like this with `TODO: `. This lets us quickly grep the code base for things that are marked in this way and lets us make sure that we've taken care of all of them before submitting. Alternatively they can be added as issues in the github repo. Likewise, if something is a fix, rather than a to do, it's common to use `FIXME: ` in an equivalent way.